lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
Here is what I could come up with:

- Leave it this way except that null value is not allowed, checking for domain node is done by calling hasNode instead of the current method (I prefer this)
- Add a conditional logic to the generated code so that dgettext is called when a valid domain is passed (@stof prefers this)

IMO the domain node works exactly the same as plural except it MUST have a value as well.

@fabpot any thoughts?

/cc @TamasBarta